### PR TITLE
Replace ask with sending request to make Mesos calls.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
The last MWT showed `AskTimeoutExceptions` when Mesos would not respond in time during
a Mesos master failover. This timeout was the result of an HTTP request timing out and is hard
to debug. It also leaves dead letters.

This change sends a promise instead. The promise is completed with the HTTP request result
which can include a timeout. It should ease debugging.

### CODE REVIEW

File: mesos-client/src/main/scala/com/mesosphere/mesos/client/Session.scala [hunk 1]
```diff
@@ -40,7 +44,15 @@ case class Session(baseUri: Uri, streamId: String, authorization: Option[Credent
       .watch(ref)
       .mapAsync(parallelism) {
         case (el, ctx) =>
-          akka.pattern.ask(ref).?(el)(timeout).mapTo[HttpResponse](tag).map(o => (o, ctx))
+          akka.pattern
+            .ask(ref)
+            .?(el)(timeout)
+            .transform {
+              case Failure(ex: AskTimeoutException) => Failure(SessionRequestException("Mesos request timed out.", ex))
```
Comment (jeschkies) [inline mesos-client/src/main/scala/com/mesosphere/mesos/client/Session.scala]: This is just illustrating an issue. The `Http().singleRequest` might not complete in time an trigger this `AskTimeoutException`. It also leaves a dead letter. I do not like it. @timcharper, how should we do this? I'd say we increase the ask time out and let Akka trigger a client timeout. I just don't know which of [these](https://doc.akka.io/docs/akka-http/current/common/timeouts.html) we have to watch for.

Comment (jeschkies) [reply mesos-client/src/main/scala/com/mesosphere/mesos/client/Session.scala]: The session actor should reply with a `Future` immediately.

Comment (jeschkies) [reply mesos-client/src/main/scala/com/mesosphere/mesos/client/Session.scala]: And convert it to a typed actor.

**Review by timcharper — APPROVED** [review]
Please consider extracting out the promise concern to a method. Looks good!

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
[info] Running scalafix on 14 Scala sources
[info] Running scalafix on 2 Scala sources
--- /home/travis/build/mesosphere/usi/mesos-client/src/test/scala/com/mesosphere/mesos/client/SessionActorTest.scala
+++ &lt;expected fix&gt;
@@ -5,11 +5,9 @@
 import akka.http.scaladsl.model._
 import akka.http.scaladsl.model.headers.{BasicHttpCredentials, HttpCredentials}
 import akka.http.scaladsl.unmarshalling.Unmarshal
-import akka.util.Timeout
 import com.mesosphere.utils.AkkaUnitTest
 
 import scala.concurrent.{Future, Promise}
-import scala.concurrent.duration._
 
 class SessionActorTest extends AkkaUnitTest {
 
[error] (mesos-client / Test / scalafix) scalafix.sbt.ScalafixFailed: TestError
[error] Total time: 4 s, completed Jun 9, 2020 12:00:50 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Pull request merged**